.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,26 @@
+name: golangci-lint
```

Comment:
wonder if it would be easier to manage if we put the jobs in the same file so we can use variables for golang version, etc.

Reply:
That might be helpful. Once the other build and test action is merged I'll rebase this one.

Reply:
The only changes that don't come from lint fixes are in `golangci.yml` and in `ci.yml`.

Reply:
Done.